core/src/test/java/org/elasticsearch/search/aggregations/metrics/percentiles/hdr/InternalHDRPercentilesRanksTests.java

```diff
@@ -61,4 +66,26 @@ protected void assertReduced(InternalHDRPercentileRanks reduced, List<InternalHD
         return InternalHDRPercentileRanks::new;
     }
 
+    @Override
+    protected NamedXContentRegistry xContentRegistry() {
+        return new NamedXContentRegistry(singletonList(
+                new NamedXContentRegistry.Entry(
+                        Aggregation.class,
+                        new ParseField(InternalHDRPercentileRanks.NAME),
+                        (parser, context) -> ParsedHDRPercentileRanks.fromXContent(parser, (String) context))));
```

Comment:
Ok. I wanted to change that in a follow up method to limit the set of changes of this PR but OK, let's do it now.